registration/backends/enhanced/views.py

```diff
@@ -0,0 +1,172 @@
+        email should contain an activation link. The email will be
+        rendered using two templates. See the documentation for
+        ``RegistrationProfile.send_activation_email()`` for
+        information about these templates and the contexts provided to
+        them.
+
+        After the ``User`` and ``RegistrationProfile`` are created and
+        the activation email is sent, the signal
+        ``registration.signals.user_registered`` will be sent, with
+        the new ``User`` as the keyword argument ``user`` and the
+        class of this backend as the sender.
+
+        """
+        site = get_current_site(self.request)
+
+        if hasattr(form, 'save'):
+            new_user_instance = form.save()
+        else:
+            new_user_instance = (UserModel().objects
+                                 .create_user(**form.cleaned_data))
+
+        new_user = RegistrationProfile.objects.create_inactive_user(
+            new_user=new_user_instance,
+            site=site,
+            send_email=self.SEND_ACTIVATION_EMAIL,
+            request=self.request,
+        )
+        signals.user_registered.send(sender=self.__class__,
+                                     user=new_user,
+                                     request=self.request)
+        return new_user
+
+    def registration_allowed(self):
+        """
+        Indicate whether account registration is currently permitted,
+        based on the value of the setting ``REGISTRATION_OPEN``. This
+        is determined as follows:
+
+        * If ``REGISTRATION_OPEN`` is not specified in settings, or is
+          set to ``True``, registration is permitted.
+
+        * If ``REGISTRATION_OPEN`` is both specified and set to
+          ``False``, registration is not permitted.
+
+        """
+        return getattr(settings, 'REGISTRATION_OPEN', True)
+
+
+class ActivationView(BaseActivationView):
+    def activate(self, *args, **kwargs):
+        """
+        Given an an activation key, look up and activate the user
+        account corresponding to that key (if possible).
+
+        After successful activation, the signal
+        ``registration.signals.user_activated`` will be sent, with the
+        newly activated ``User`` as the keyword argument ``user`` and
+        the class of this backend as the sender.
+
+        """
+        activation_key = kwargs.get('activation_key', '')
+        activated_user = RegistrationProfile.objects.activate_user(
+            activation_key,
+            get_current_site(self.request),
+        )
+        if activated_user:
+            signals.user_activated.send(sender=self.__class__,
+                                        user=activated_user,
+                                        request=self.request)
+        return activated_user
+
+    def get_success_url(self, user):
+        return ('registration_activation_complete', (), {})
+
+
+class ApprovalView(BaseApprovalView):
```

Comment:
pending @dicato comments about the urls file, i believe that all views except this one should be inherited from the `default` backend. There is no need to repeat this code and then have to propagate bug fixes between the two.